packages/react-core/src/components/TimePicker/TimePicker.tsx

```diff
@@ -401,107 +423,68 @@ export class TimePicker extends React.Component<TimePickerProps, TimePickerState
     const options = makeTimeOptions(stepMinutes, !is24Hour, delimiter, minTimeState, maxTimeState);
     const isValidFormat = this.isValidFormat(timeState);
     const randomId = id || getUniqueId('time-picker');
 
     const menuContainer = (
-      <ul
-        ref={this.menuRef}
-        className={css(styles.selectMenu)}
-        role="listbox"
-        aria-labelledby={`${id}-input`}
-        style={{ maxHeight: '200px', overflowY: 'auto' }}
-      >
-        {options.map((option, index) => (
-          <TimeOption
-            key={index}
-            value={option}
-            index={index}
-            onSelect={this.onSelect}
-            isFocused={index === focusedIndex}
-            id={`${id}-option-${index}`}
-          />
-        ))}
-      </ul>
-    );
-
-    const inputAndToggle = (
-      <div className={css(datePickerStyles.datePickerInput)} style={style} {...props}>
-        <InputGroup>
-          <div
-            className={css(
-              styles.select,
-              isOpen && styles.modifiers.expanded,
-              direction === SelectDirection.up && styles.modifiers.top,
-              className
-            )}
-            id={randomId}
-            ref={this.parentRef}
-          >
-            <div
-              ref={this.toggleRef}
-              className={css(styles.selectToggle, isDisabled && styles.modifiers.disabled, styles.modifiers.typeahead)}
-              style={{ paddingLeft: '0' }}
-            >
-              <TextInput
-                className={css(formStyles.formControl, styles.selectToggleTypeahead)}
-                id={`${randomId}-input`}
-                aria-label={ariaLabel}
-                validated={isInvalid ? 'error' : 'default'}
-                placeholder={placeholder}
-                value={timeState || ''}
-                type="text"
-                iconVariant="clock"
-                onClick={this.onInputFocus}
-                onFocus={this.onInputFocus}
-                onChange={this.onInputChange}
-                onBlur={this.onBlur}
-                autoComplete="off"
-                isDisabled={isDisabled}
-                ref={this.inputRef}
-                {...inputProps}
-              />
-            </div>
-            {isOpen && menuAppendTo === 'inline' && menuContainer}
-          </div>
-        </InputGroup>
-        {isInvalid && (
-          <div className={css(datePickerStyles.datePickerHelperText, datePickerStyles.modifiers.error)}>
-            {!isValidFormat ? invalidFormatErrorMessage : invalidMinMaxErrorMessage}
-          </div>
-        )}
-      </div>
+      <Menu ref={this.menuRef} isScrollable>
+        <MenuContent maxMenuHeight="200px">
+          <MenuList aria-labelledby={`${randomId}-input`}>
+            {options.map((option, index) => (
+              <MenuItem onClick={this.onSelect} key={index} id={`${randomId}-option-${index}`}>
```

Comment:
Using the randomId or the whole string used for `id` (could be stored as a constant and used in both places) would be more reliable as the value for `key`.